miso-service/src/main/java/uk/ac/bbsrc/tgac/miso/service/impl/DefaultBoxService.java

```diff
@@ -0,0 +1,311 @@
+import java.util.function.Consumer;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+
+import com.google.common.collect.Sets;
+
+import uk.ac.bbsrc.tgac.miso.core.data.AbstractBox;
+import uk.ac.bbsrc.tgac.miso.core.data.Box;
+import uk.ac.bbsrc.tgac.miso.core.data.BoxSize;
+import uk.ac.bbsrc.tgac.miso.core.data.BoxUse;
+import uk.ac.bbsrc.tgac.miso.core.data.impl.changelog.BoxChangeLog;
+import uk.ac.bbsrc.tgac.miso.core.data.impl.view.BoxableView;
+import uk.ac.bbsrc.tgac.miso.core.data.impl.view.BoxableView.BoxableId;
+import uk.ac.bbsrc.tgac.miso.core.exception.MisoNamingException;
+import uk.ac.bbsrc.tgac.miso.core.service.naming.NamingScheme;
+import uk.ac.bbsrc.tgac.miso.core.store.BoxStore;
+import uk.ac.bbsrc.tgac.miso.core.store.ChangeLogStore;
+import uk.ac.bbsrc.tgac.miso.core.store.SecurityProfileStore;
+import uk.ac.bbsrc.tgac.miso.core.util.PaginatedDataSource;
+import uk.ac.bbsrc.tgac.miso.core.util.PaginationFilter;
+import uk.ac.bbsrc.tgac.miso.service.BoxService;
+import uk.ac.bbsrc.tgac.miso.service.security.AuthorizationManager;
+import uk.ac.bbsrc.tgac.miso.service.security.AuthorizedPaginatedDataSource;
+
+@Service
+public class DefaultBoxService implements BoxService, AuthorizedPaginatedDataSource<Box> {
+  @Autowired
+  private AuthorizationManager authorizationManager;
+  @Value("${miso.autoGenerateIdentificationBarcodes}")
+  private Boolean autoGenerateIdBarcodes;
+
+  @Autowired
+  private BoxStore boxStore;
+
+  @Autowired
+  private ChangeLogStore changeLogStore;
+
+  @Autowired
+  private NamingScheme namingScheme;
+
+  @Autowired
+  private SecurityProfileStore securityProfileStore;
+  private void addBoxContentsChangeLog(Box box, String message) throws IOException {
+    BoxChangeLog changeLog = new BoxChangeLog();
+    changeLog.setBox(box);
+    changeLog.setTime(new Date());
+    changeLog.setColumnsChanged("contents");
+    changeLog.setUser(authorizationManager.getCurrentUser());
+    changeLog.setSummary(message);
+    changeLogStore.create(changeLog);
+  }
+
+  private void applyChanges(Box from, Box to) throws IOException {
+    to.setAlias(from.getAlias());
+    to.setDescription(from.getDescription());
+    to.setIdentificationBarcode(from.getIdentificationBarcode());
+    to.setUse(boxStore.getUseById(from.getUse().getId()));
+  }
+
+  @Override
+  public long count(Consumer<String> errorHandler, PaginationFilter... filter) throws IOException {
+    return boxStore.count(errorHandler, filter);
+  }
+
+  @Override
+  public void deleteBox(Box box) throws IOException {
+    authorizationManager.throwIfNotWritable(box);
+    if (boxStore != null) {
+      if (!boxStore.remove(box)) {
+        throw new IOException("Unable to delete box.");
+      }
+    } else {
+      throw new IOException("No boxStore available. Check that it has been declared in the Spring config.");
```

Comment:
Could you please remove the checks for boxStore != null ? The lack of this particular cruft is one of my favourite parts of the services.